openassessment/xblock/static/js/src/studio/oa_edit_prompts.js

```diff
@@ -13,15 +13,25 @@ OpenAssessment.EditPromptsView = function(element, notifier) {
     this.editorElement = $(this.element).closest("#openassessment-editor");
     this.addRemoveEnabled = this.editorElement.attr('data-is-released') !== 'true';
 
+    var addWysiwyg = window.tinyMCE !== undefined;
```

Comment:
Why do you calculate `addWysiwyg` at this point? Could you check for tinyMCE in `WysiwygPrompt`?

You don't have to pass the `addWysiwyg` to `WysiwygPrompt` if it can check for the value internally.

Reply:
@sandroroux i'm not quite sure that understand you, You asked me not to modify "element" in the constructor:

>The Prompt constructor should not be modifying "element". Looking at the constructor's original code, it expects to receive its "element" from wherever Prompt is instantiated.
>
>My recommendations:
>
>Wrap this code in a helper method (lines 83-88).
>Modify "element", then pass its modified value into the Prompt constructor. We construct Prompt in "OpenAssessment.EditPromptsView".

So I've created special function [prepareElement](https://github.com/edx/edx-ora2/pull/1058/files#diff-d3ccf467bd20f391e1bb1d3a4298b1b0R27) where I need to detect is tinyMCE available or not. 

Besides this we should know what type of element should be used: [wysiwyg or textarea](https://github.com/edx/edx-ora2/pull/1058/files#diff-d3ccf467bd20f391e1bb1d3a4298b1b0R19)

Reply:
I'm sorry for the confusion. Let me explain myself. 

While I like the idea, I don't think you need to make a `WysiwygPrompt` class. It doesn't specialize `Prompt` enough, and I do not want to deepen the prototype tree. You should revert to your original `Prompt`-based implementation.

From what I see, `Prompt` can become a wysiwyg editor with two changes:
1. Attach tinyMCE to `element`, and pass it to the `Prompt` constructor.
2. Update `Prompt`'s `description` to pull content from the tinyMCE element.

You can eliminate `addWysiwyg` if you compute its value where you use it (at changes 1 and 2 – the bodies of `prepareElement` and `Prompt::description`). Computing it _in situ_ frees you from having to pass it to `Prompt` in a `kwargs` object. Removing branching on `addWysiwyg` will clean up the code.

Please let me know if my explanation doesn't make sense.